packages/common/src/ui/components/inputs/Autocomplete/utils.tsx

```diff
@@ -9,12 +9,12 @@ export const DefaultAutocompleteItemOption = styled('li')(({ theme }) => ({
 
 type Option<T> = { label: string } & T;
 
-export type OptionMapper<T extends Record<keyof T, unknown>> = (
+export type OptionMapper<T extends { [K in keyof T]: T[K] }> = (
   options: T[],
   key: keyof T
 ) => Option<T>[];
 
-export const defaultOptionMapper = <T extends Record<keyof T, unknown>>(
+export const defaultOptionMapper = <T extends { [K in keyof T]: T[K] }>(
```

Comment:
this, after removing the `ObjectWithStringKeys` was having a few problems with things being passed in. This type is 'copying' over the same type from T, so optionals are carried over, where as `Record<keyof T, unknown>` was making it so every key was non optional

Reply:
that's very neat!